### PR TITLE
Adds generation expression to MySQL57 getListTableColumns method

### PR DESCRIPTION
**Pull request by octavianparalescu**
The generation expression is very important for the table columns, it is actually the definition of virtual columns.

https://www.percona.com/blog/2016/03/04/virtual-columns-in-mysql-and-mariadb/

### CODE REVIEW

**Comment by Ocramius** [comment]
Requires integration test additions

**Comment by OctavianParalescu** [comment]
@Ocramius Unfortunately I cannot seem to be able to add virtual columns using \Doctrine\DBAL\Schema\Column 's option array, it seems we are completely lacking virtual/stored columns support from MySQL 5.7.

**Comment by Ocramius** [comment]
I'd run the DDL in a raw query, then verify what the introspection does.

File: lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php [hunk 1]
```diff
@@ -204,6 +204,18 @@ protected function _getPortableTableColumnDefinition($tableColumn)
             $column->setPlatformOption('collation', $tableColumn['collation']);
         }
 
+        // MySQL 5.7 specific https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html
+        if (strpos($tableColumn['extra'], 'GENERATED') !== false) {
+            // extra's syntax in the case of a generated columns:
+            // <UPPERCASE_GENERATION_TYPE> GENERATED
+            $pieces = explode(' ', $tableColumn['extra']);
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php]: You only use `$pieces[0]` here. Maybe you'd rather use `[$generationType] = explode(' ', $tableColumn['extra']);`? Also, is there a limit in the `explode()` expression here? Do you care only about the first space? If so, add `2` as a parameter to `explode()`

**Review by Ocramius — DISMISSED** [review]
This looks good to me, but I'm unsure about MySQL 5.7-specific details in the schema manager rather than in the platform. @morozov where do you think this would fit best?

File: lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php (hunk 1 above)
Comment (morozov) [reply lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php]: Or `$generationType = strstr($tableColumn['extra'], ' ', true);`?

**Comment by morozov** [comment]
> I'm unsure about MySQL 5.7-specific details in the schema manager rather than in the platform. @morozov where do you think this would fit best?

It depends on what the separation of concerns between the platform and schema manager is. I think we can merge it as is and think it through later. E.g., having a MariaDB-specific code in MySQL schema manager is a bigger concern to me.

**Comment by OctavianParalescu** [comment]
We also need to modify the docs at http://docs.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/schema-representation.html (10.1.1.3. Vendor specific options')

**Pull request closed without merging**